### PR TITLE
update containerd to v1.2.1

### PR DESCRIPTION
**Pull request by andrewhsu**
Update to latest containerd 1.2.1 release https://github.com/containerd/containerd/releases/tag/v1.2.1

Also updated moby's `vendor.conf` with changes in containerd's `vendor.conf`. Link to upstream differences:
* https://github.com/containerd/containerd/compare/de1f167...9b32062
* https://github.com/opencontainers/runc/compare/10d38b6...96ec217
* https://github.com/containerd/cri/compare/f913714...0ca1e3c

I mostly modeled this PR after #38245

### CODE REVIEW

**Comment by andrewhsu** [comment]
Hmm..couple of [errors](https://jenkins.dockerproject.org/job/Docker-PRs-experimental/43136/console):
```
# github.com/docker/docker/pkg/sysinfo
pkg/sysinfo/sysinfo_linux.go:235:40: not enough arguments in call to cgroups.FindCgroupMountpoint
	have (string)
	want (string, string)
# github.com/docker/docker/vendor/github.com/docker/libnetwork
vendor/github.com/docker/libnetwork/sandbox_externalkey_unix.go:58:12: undefined: configs.HookState
```

Looks like function signature changed from https://github.com/opencontainers/runc/pull/1872

@thaJeztah @crosbymichael @kolyshkin any advice on how to account for the new parameter? (feel free to push to this PR)

**Comment by crosbymichael** [comment]
Update the code as well

**Comment by andrewhsu** [comment]
I've added commit to pass empty string as `cgroupPath` argument to `cgroups.FindCgroupMountpoint` to mimic previous behaviour.

**Comment by kolyshkin** [comment]
> 00:09:54.531 vendor/github.com/docker/libnetwork/sandbox_externalkey_unix.go:58:12: undefined: configs.HookState

**Comment by kolyshkin** [comment]
Cause is https://github.com/opencontainers/runc/commit/e23868603a8bac8f374f5bca129e7ed187cd51c9
Fix seems to be similar

**Comment by thaJeztah** [comment]
PowerPC; https://jenkins.dockerproject.org/job/Docker-PRs-powerpc/12354/console

```
03:31:45 FAIL: docker_api_swarm_node_test.go:75: DockerSwarmSuite.TestAPISwarmNodeDrainPause
03:31:45 
03:31:45 [d767e4e600bfa] waiting for daemon to start
03:31:45 [d767e4e600bfa] daemon started
03:31:45 
03:31:45 [d6769e6520ba3] waiting for daemon to start
03:31:45 [d6769e6520ba3] daemon started
03:31:45 
03:31:45 waited for 1.976510522s (out of 30s)
03:31:45 waited for 66.59312ms (out of 30s)
03:31:45 waited for 963.341376ms (out of 30s)
03:31:45 waited for 3.034784534s (out of 30s)
03:31:45 waited for 19.7508ms (out of 30s)
03:31:45 docker_api_swarm_node_test.go:104:
03:31:45     waitAndAssert(c, defaultReconciliationTimeout*2, reducedCheck(sumAsIntegers, d1.CheckActiveContainerCount, d2.CheckActiveContainerCount), checker.Equals, instances)
03:31:45 docker_utils_test.go:441:
03:31:45     c.Assert(v, checker, args...)
03:31:45 ... obtained int = 3
03:31:45 ... expected int = 1
03:31:45 ... output: "8526f9d920ac\n48aab63974a9\n1bcc0e3001ef\n"
03:31:45 
03:31:45 waited for 1m0.030588516s (out of 1m0s)
03:31:45 [d767e4e600bfa] exiting daemon
03:31:45 [d6769e6520ba3] exiting daemon
```

**Comment by thaJeztah** [comment]
Rebased to get rid of the containerd/runc binary bumps (which are already merged as part of https://github.com/moby/moby/pull/38328, but CI was green before that;

screen shot 2018-12-07 at 18 21 23

File: vendor.conf [hunk 1]
```diff
@@ -37,7 +37,7 @@ github.com/mitchellh/hashstructure 2bca23e0e452137f789efbc8610126fd8b94f73b
 #get libnetwork packages
 
 # When updating, also update LIBNETWORK_COMMIT in hack/dockerfile/install/proxy.installer accordingly
-github.com/docker/libnetwork 1f28166bb386cf9223d2d00a28382b0e474be314
+github.com/docker/libnetwork 7b76eca2385c470f850d942c4a9dfaadf945ed0e https://github.com/andrewhsu/libnetwork
```
Comment (thaJeztah) [inline vendor.conf]: oh! have to update this one, because this was merged

**Comment by crosbymichael** [comment]
i'll fix the hook stuff today for ya

**Comment by thaJeztah** [comment]
Windows RS5 failure https://jenkins.dockerproject.org/job/Docker-PRs-WoW-RS5-Process/586/console should be unrelated, but let me restart;

```
17:56:48 FAIL: docker_api_images_test.go:60: DockerSuite.TestAPIImagesSaveAndLoad
17:56:48 
17:56:48 assertion failed: 
17:56:48 Command:  d:\CI\CI-78045a5419\binary\docker.exe inspect --format {{.Id}} sha256:97b8a0c1157cc6f615a17a77c1e8be28e8f67fd1aafd693d51a6b345ce2f86ae
17:56:48 ExitCode: 1
17:56:48 Error:    exit status 1
17:56:48 Stdout:   
17:56:48 
17:56:48 Stderr:   Error: No such object: sha256:97b8a0c1157cc6f615a17a77c1e8be28e8f67fd1aafd693d51a6b345ce2f86ae
17:56:48 
17:56:48 
17:56:48 Failures:
17:56:48 ExitCode was 1 expected 0
17:56:48 Expected no error
17:56:49 
```

**Comment by thaJeztah** [comment]
PowerPC and Z failures are known flaky; tracked through https://github.com/moby/moby/issues/32673, and being worked on in https://github.com/moby/moby/pull/37833

```
19:12:33 FAIL: docker_api_swarm_test.go:296: DockerSwarmSuite.TestAPISwarmLeaderElection
19:12:33 
19:12:33 [d799fa85b70bf] waiting for daemon to start
19:12:33 [d799fa85b70bf] daemon started
19:12:33 
19:12:33 [d384dfce963a1] waiting for daemon to start
19:12:33 [d384dfce963a1] daemon started
19:12:33 
19:12:33 [defa9a5347d07] waiting for daemon to start
19:12:33 [defa9a5347d07] daemon started
19:12:33 
19:12:33 [d799fa85b70bf] exiting daemon
19:12:33 Waiting for election to occur...
19:12:33 assertion failed: error is not nil: Error response from daemon: rpc error: code = DeadlineExceeded desc = context deadline exceeded: [d384dfce963a1] (*Daemon).GetNode: NodeInspectWithRaw("nxems896l6insqhmydvdd16pf") failed
19:12:33 waited for 20.002158381s (out of 30s)
19:12:33 [d384dfce963a1] exiting daemon
19:12:33 [defa9a5347d07] exiting daemon
```

**Comment by andrewhsu** [comment]
@thaJeztah your updates to this PR LGTM

**Comment by crosbymichael** [comment]
LGTM

**Review by tiborvass — APPROVED** [review]

**Pull request merged**